packages/react/src/components/controlComponents.tsx

```diff
@@ -41,22 +46,75 @@ export const ClerkLoading = ({ children }: React.PropsWithChildren<unknown>): JS
   return <>{children}</>;
 };
 
-type GateProps = React.PropsWithChildren<
-  Parameters<experimental__CheckAuthorizationWithoutPermission>[0] & {
+type ProtectProps = React.PropsWithChildren<
+  (
+    | {
+        condition?: never;
+        role: OrganizationCustomRoleKey;
+        permission?: never;
+      }
+    | {
+        condition?: never;
+        role?: never;
+        permission: OrganizationCustomPermissionKey;
+      }
+    | {
+        condition: (has: CheckAuthorizationWithCustomPermissions) => boolean;
+        role?: never;
+        permission?: never;
+      }
+    | {
+        condition?: never;
+        role?: never;
+        permission?: never;
+      }
+  ) & {
     fallback?: React.ReactNode;
   }
 >;
 
 /**
- * @experimental The component is experimental and subject to change in future releases.
+ * Use `<Protect/>` in order to prevent unauthenticated or unauthorized user from accessing the children passed to the component.
```

Comment:
```suggestion
 * Use `<Protect/>` in order to prevent unauthenticated or unauthorized users from accessing the children passed to the component.
```